sdk/dotnet/Pulumi/Resources/CustomResource.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016-2019, Pulumi Corporation
+
+#nullable enable
+
+using Pulumi.Rpc;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// CustomResource is a resource whose create, read, update, and delete(CRUD) operations are
+    /// managed by performing external operations on some physical entity. The engine understands
+    /// how to diff and perform partial updates of them, and these CRUD operations are implemented
+    /// in a dynamically loaded plugin for the defining package.
+    /// </summary>
+    public class CustomResource : Resource
+    {
+        internal readonly IdOutputCompletionSource _id;
+
+        /// <summary>
+        /// Id is the provider-assigned unique ID for this managed resource.  It is set during
+        /// deployments and may be missing (unknown) during planning phases.
+        /// </summary>
+        public Output<Id> Id => _id.Output;

```

Comment:
Some resources (e.g. Azure App Service) expect IDs of other resources (App Service Plan) as the value of their inputs (`AppServicePlanId`), and it's of type string. Are those the same values? If so, we need a clean way to extract the value from `Id` type.

Reply:
Good question.

Reply:
I checked and they are the same, but now I have to `Substring` of `ToString` to get the value...

Reply:
these are now strings.